src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -601,9 +609,9 @@ private void Cleanup()
                     if (_httpRequestMessage != null)
                     {
                         _timeoutHelper.CancelCancellationToken(false);
-                        this.TryCompleteHttpRequest(_httpRequestMessage);
+                        TryCompleteHttpRequest(_httpRequestMessage);
+                        _httpRequestMessage.Dispose();
                     }
```

Comment:
TryCompleteHttpRequest currently doesn't do anything. It uses an interlocked compare exchange to ensure _channel.OnHttpRequestCompleted is only called once. OnHttpRequestCompleted was used for cleaning up some native client certificate security allocations that we don't do at the moment. That code shouldn't throw. If a change in the future is made such that it can and does throw, the worst that could happen is the stream and allocated buffer in the request object won't get cleaned up until the GC runs and the buffer won't get returned to the buffer pool.